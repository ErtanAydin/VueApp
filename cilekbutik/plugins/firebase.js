import * as firebase from 'firebase/app'
import 'firebase/auth'

const firebaseConfig = {
    apiKey: "AIzaSyC7kInVEj6L35XB8NX_iwdgKvpnzpkK_WU",
    authDomain: "cilekbutik-9efd9.firebaseapp.com",
    projectId: "cilekbutik-9efd9",
    storageBucket: "cilekbutik-9efd9.appspot.com",
    messagingSenderId: "128564693192",
    appId: "1:128564693192:web:34362c3303bb2a162b3dfd",
    measurementId: "G-V1W8HH0S1K"
  };

  let app = null;
  if(!firebase.apps.length){
      app = firebase.initializeApp(firebaseConfig);
  }

export default firebase;