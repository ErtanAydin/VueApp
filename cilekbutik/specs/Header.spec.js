import { mount, RouterLinkStub } from "@vue/test-utils";
import Header from "@/components/Header";
import * as firebase from 'firebase'

const onAuthStateChanged = jest.fn()

const getRedirectResult = jest.fn(() => {
  return Promise.resolve({
    user: {
      email:'a@hotmail.com',
      sifre: '123456'
    }
  })
})


const initializeApp = jest
  .spyOn(firebase, 'initializeApp')
  .mockImplementation(() => {
    return {
      auth: () => {
        return {
          getRedirectResult
        }
      }
    }
  })

  jest.spyOn(firebase, 'auth').mockImplementation(() => {
    return {
      onAuthStateChanged,
      currentUser: {
        email: 'a@hotmail.com',
        sifre: '123456'
      },
      getRedirectResult,

    }
  })

describe("mounted with firebase", () => {
  test("mounted with firebase test", async () => {
    const wrapper = mount(Header, {
      stubs: {
        NuxtLink: RouterLinkStub
      }});
      
      expect(wrapper.vm.user).toBe(wrapper.vm.user)
      expect(firebase.auth).toHaveBeenCalled()
      if(wrapper.vm.user=null){ 
        expect(wrapper.vm.UyeOlText).toBe('Çıkış Yap')
        expect(wrapper.vm.GirisYapText).toBe('Hesabım')
      }
  })
})

describe("methods", () => {
  test("signout", async () => {
    const mockRouter = {
      push: jest.fn()
    }
    const wrapper = mount(Header, {
      props: {
        authenticated: true
      },
      global: {
        mocks: {
          $router: mockRouter
        }
      },
      stubs: {
        NuxtLink: RouterLinkStub,
      }});
      wrapper.vm.email = ''
      wrapper.vm.sifre = ''
      expect(firebase.auth).toHaveBeenCalled()
      expect(wrapper.vm.user).toBe('')
      expect(wrapper.vm.UyeOlText).toBe('Üye Ol')
      expect(wrapper.vm.GirisYapText).toBe('Giriş Yap')
      expect(mockRouter.push)
      expect(console.clear())
      expect(console.log('Cikis yapildi.') )
  })
})

describe("Header", () => {
  test("mounts properly", () => {
    const wrapper = Header;
    expect(wrapper).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = mount(Header, {
      stubs: {
        NuxtLink: RouterLinkStub
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe ('data', () => {
  test('data test', () => {
    const wrapper = mount(Header,{
    stubs: {
      NuxtLink: RouterLinkStub
    }});
    expect(wrapper.vm.user).toBe('')
    expect(wrapper.vm.eposta).toBe('')
    expect(wrapper.vm.sifre).toBe('')
    expect(wrapper.vm.UyeOlText).toBe('Üye Ol')
    expect(wrapper.vm.GirisYapText).toBe('Giriş Yap')
  })
})