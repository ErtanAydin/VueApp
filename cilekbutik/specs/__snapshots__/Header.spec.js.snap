// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
<header>
  <section class="duyuru">
    <div class="duyuru-text"> 200 TL VE ÜZERİ SİPARİŞLERDE ÜCRETSİZ KARGO FIRSATI SİZLERİ BEKLİYOR </div>
  </section>
  <div class="header">
    <section class="ana-bolum">
      <div class="logo-bolumu"><a><img src="https://www.cilekbutik.net/images/logo/logo-16.png" alt=""></a></div>
      <div class="orta-bolum">
        <form action="https://www.cilekbutik.net/arama" class="searchbar-form"><select name="kategoriler" class="kategori-selectbox">
            <option value="0">Tüm Kategoriler</option>
            <option value="549">Üst Giyim</option>
            <option value="550">Alt Giyim</option>
            <option value="551">Dış Giyim</option>
            <option value="552">Elbiseler</option>
            <option value="553">Aksesuar</option>
            <option value="554">Pijama Takımı</option>
            <option value="555">İkili takım</option>
            <option value="573">İndirimli Ürünler</option>
          </select>
          <div class="searchbar-tasarim"></div> <input type="text" name="searchbar-input-text" placeholder="Aramak istediğiniz ürünü yazın..." class="searchbar-inputbox">
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"> <button type="submit" class="searchbar-input-button"><i class="fa fa-search search-icon" style="font-size: large;"></i></button>
        </form>
      </div>
      <div class="yan-bolum">
        <div class="yan-bolum-icerik">
          <div class="tepe-giris-uyeol">
            <div class="uyelik-linkler"><a class="tepe-link-text">Giriş Yap</a> <span> / </span> <a href="#" class="tepe-link-text">Üye Ol</a></div>

            <i class="fa fa-user kullanici-icon"></i>
          </div> <a class="tepe-sepet"><span class="tepe-sepet-bilgi"><span class="sepetim-text">Sepetim</span> <br> <span class="sepetim-ucret-text">50,00 TL</span></span> <span class="tepe-sepet-icon-bar"><i class="fa fa-shopping-basket tepe-sepet-icon"></i> <span class="tepe-urun-adedi">1</span></span></a>
        </div>
      </div>
    </section>
  </div>
</header>
`;
